woocommerce/woocommerce-android
# Sync WPCOM status with Privacy Banner state

Closes: #8947


### Description

This PR syncs adds behaviour of showing the recent state of user choice (from WPCOM) on privacy banner.
### Testing instructions

1. Log in via WPCOM account and apply PATCH from the bottom of the description
2. Turn on privacy settings on https://wordpress.com/me/privacy
3. Run the app. Assert you can see the privacy banner with Analytics enabled
4. Turn off privacy settings on https://wordpress.com/me/privacy
5. Run the app. Assert you can see the privacy banner with Analytics disabled


<details><summary>PATCH</summary>

```PATCH
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt	(revision 517ad8f082b0f64343f9974a687d90161e4199c6)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt	(date 1684421458319)
@@ -129,10 +129,10 @@
         }
         observeTopPerformerUpdates()
 
+        triggerEvent(MyStoreEvent.ShowPrivacyBanner)
         if (FeatureFlag.PRIVACY_CHOICES.isEnabled()) {
             shouldShowPrivacyBanner().let {
                 if (it) {
-                    triggerEvent(MyStoreEvent.ShowPrivacyBanner)
                 }
             }
         }
```
</details>
